src/scales/scale.time.js

```diff
@@ -662,11 +662,15 @@ module.exports = Scale.extend({
 		var majorUnit = me._majorUnit;
 		var majorFormat = formats[majorUnit];
 		var majorTime = +adapter.startOf(time, majorUnit);
-		var majorTickOpts = options.ticks.major;
+		var tickOpts = options.ticks;
+		var majorTickOpts = tickOpts.major;
 		var major = majorTickOpts.enabled && majorUnit && majorFormat && time === majorTime;
 		var label = adapter.format(time, format ? format : major ? majorFormat : minorFormat);
-		var tickOpts = major ? majorTickOpts : options.ticks.minor;
-		var formatter = valueOrDefault(tickOpts.callback, tickOpts.userCallback);
+		var nestedTickOpts = major ? majorTickOpts : tickOpts.minor;
+		var formatter = helpers.options.resolve([
```

Comment:
```javascript
var formatter = helpers.options.resolve([
    nestedTickOpts.callback,
    nestedTickOpts.userCallback,
    tickOpts.callback,
    tickOpts.userCallback
]);
```

But do we really want to nest `callback` and `userCallback` under minor/major? I think the ticks array already contain this information, so we may not want to add more complexity to this minor/major (IMO wrong) logic.

```
callback: function(label, index, ticks) {
    return ticks[index].major ? /* this is a major tick */ : /* this is a minor tick */;
}
```

Reply:
Currently, core.scale is calling this callback [passing `(scale.ticks[i], i, scale.ticks)`](https://github.com/chartjs/Chart.js/blob/b9290a20de0ac051b80eca3509163d597ecec476/src/core/core.scale.js#L315), but scale.time is [passing `(scale._ticks[i].value, i, scale._ticks)`](https://github.com/chartjs/Chart.js/blob/b9290a20de0ac051b80eca3509163d597ecec476/src/scales/scale.time.js#L679). This seems a regression when the `scale._ticks` object is introduced.

I agree that `callback` and `userCallback` don't need to be nested if the `*.major` property is available, but I think the 3rd argument of them should also be consistent.

Reply:
I think the time scale always passed the tick value. I agree for the consistency but not sure we can change this in a non breaking manner?

Reply:
Oh, I didn't realize that the callbacks was already nested under minor / major, I thought it was something new with this PR. So I agree, we can't remove them.

Reply:
Ok, I just updated the description of the callback.

Reply:
Maybe also add TODO so that we don't forget to clean it up in v3